nuget/spec/dependabot/nuget/file_parser/project_file_parser_spec.rb

```diff
@@ -870,8 +918,9 @@ def dependencies_from(dep_info)
             end
 
             it "has the right details" do
-              query_stub = stub_search_results_with_versions_v3("microsoft.extensions.dependencymodel_cached",
-                                                                ["1.1.1", "1.1.0"])
+              versions_stub = stub_versions_v3("microsoft.extensions.dependencymodel_cached", ["1.1.1", "1.1.0"])
+              stub_registry_v3("microsoft.extensions.dependencymodel_cached", ["1.1.1", "1.1.0"])
```

Comment:
Does it need to be stubbed twice?

Reply:
This shouldn't stub anything twice, it stubs the call to the versions API and then the call to the registry API. the Versions API is used to quickly and cheaply establish if the package exists at all, and the registry API is used to generate the list of actual versions we need (the versions API includes unlisted versions which we need to exclude).